nvflare/job_config/base_script_runner.py

```diff
@@ -0,0 +1,242 @@
+        else the provided executor will be used.
+
+        If some components are passed in, it is user's responsibility to make sure they are consistent with each other.
+        For example, if user provide a task_pipe, the default task_pipe_id is "task_pipe",
+        please make sure the ClientAPILauncherExecutor is created using the matching pipe_id.
+
+        Args:
+            script (str): Script to run. For in-process must be a python script path. For ex-process can be any script support by `command`.
+            script_args (str): Optional arguments for script (appended to script).
+            launch_external_process (bool): Whether to launch the script in external process. Defaults to False.
+            command (str): If launch_external_process=True, command to run script (preprended to script). Defaults to "python3".
+            framework (str): Framework type to connfigure converter and params exchange formats. Defaults to FrameworkType.PYTORCH.
+            params_transfer_type (str): How to transfer the parameters. FULL means the whole model parameters are sent.
+                DIFF means that only the difference is sent. Defaults to TransferType.FULL.
+        """
+        self._script = script
+        self._script_args = script_args
+        self._command = command
+        self._launch_external_process = launch_external_process
+        self._framework = framework
+        self._params_transfer_type = params_transfer_type
+
+        self._params_exchange_format = None
+
+        if self._framework == FrameworkType.PYTORCH:
+            _, torch_ok = optional_import(module="torch")
+            if torch_ok:
+                self._params_exchange_format = ExchangeFormat.PYTORCH
+            else:
+                raise ValueError("Using FrameworkType.PYTORCH, but unable to import torch")
+        elif self._framework == FrameworkType.TENSORFLOW:
+            _, tf_ok = optional_import(module="tensorflow")
+            if tf_ok:
+                self._params_exchange_format = ExchangeFormat.NUMPY
+            else:
+                raise ValueError("Using FrameworkType.TENSORFLOW, but unable to import tensorflow")
+        elif self._framework == FrameworkType.NUMPY:
+            self._params_exchange_format = ExchangeFormat.NUMPY
+        elif self._framework == FrameworkType.RAW:
+            self._params_exchange_format = ExchangeFormat.RAW
+        else:
+            raise ValueError(f"Framework {self._framework} unsupported")
+
+        if metric_pipe is not None:
+            validate_object_for_job("metric_pipe", metric_pipe, Pipe)
+        if metric_relay is not None:
+            validate_object_for_job("metric_relay", metric_relay, MetricRelay)
+        if task_pipe is not None:
+            validate_object_for_job("task_pipe", task_pipe, Pipe)
+        if executor is not None:
+            validate_object_for_job("executor", executor, Executor)
```

Comment:
maybe we could add a more specific check for executor being a `ClientAPILauncherExecutor` or `InProcessClientAPIExecutor` depending on `launch_external_process`

Reply:
thanks! added as suggested.